Branch-SDK/src/io/branch/referral/ServerRequestQueue.java

```diff
@@ -105,7 +104,10 @@ public void run() {
                     JSONObject json = jsonArr.getJSONObject(i);
                     ServerRequest req = ServerRequest.fromJSON(json, context);
                     if (req != null) {
-                        result.add(req);
+                        // No need to retrieve close or logout from previous session
+                        if (!(req instanceof ServerRequestRegisterClose || req instanceof ServerRequestLogout)) {
```

Comment:
quick q here. in the case of a logout not being queued up to be fired, we'd still use an old identity ID inside the following v1/open right?